pytripgui/canvas_vc/canvas_controller.py

```diff
@@ -38,19 +43,20 @@ def on_mouse_wheel(self, event):
         else:
             self._model.projection_selector.prev_slice()
 
-        self.update_canvas_view()
-        self._ui.update()
+        # scrolling increments or decrements current slice number which is used to set value on slider
+        # and that slider emits event that invokes callback - set_current_slice_no
+        self._ui.position = self._model.projection_selector.current_slice_no
 
     def set_current_slice_no(self, slice_no):
         self._model.projection_selector.current_slice_no = slice_no
-
         self.update_canvas_view()
         self._ui.update()
 
     def clear_view(self):
         self._ui.clear()
 
     def update_canvas_view(self):
+        print('slice number currently processed in update: ', self._model.projection_selector.current_slice_no)
```

Comment:
Use logger (info or debug) to avoid printing to users.

Reply:
That will cut out after testing is done